override translations (#114)
glpi-project/doc: source/advanced/override-locales.rst

Override GLPI locales
---------------------

> GLPI uses `gettext` tool to manage its translations. You can override the default.

Reviewer: ```suggestion
GLPI uses `gettext <https://www.gnu.org/software/gettext/>`_ tool to manage its translations. You can override the default.
```

> GLPI uses `gettext` tool to manage its translations. You can override the default.

Reviewer 2: ```suggestion
GLPI uses the `gettext <https://www.gnu.org/software/gettext/>`_ tool to manage its translations. You can override the default translations.
```

> This is the preferred method rather editing the original po files and avoid losing your changes when updating GLPI.

Reviewer 2: ```suggestion
This is the preferred method rather than editing the original po files and risk losing your changes when updating GLPI.
```

> To do so, you need to create a file named `filename.po` in the `files/_locales/core/` directory.

Reviewer: ```suggestion
To do so, you need to create a file named ``filename.po`` in the ``files/_locales/core/`` directory.
```

> You can also create overrides for plugins with the same method in `files/_locales/pluginkey/` directory where `pluginkey` matches the plugin folder name.

Reviewer: ```suggestion
You can also create overrides for plugins with the same method in ``files/_locales/pluginkey/`` directory where ``pluginkey`` matches the plugin folder name.
```

> Depending on your GLPI configuration, the _locales directory can be located elsewhere (if you provided a `local_define.php` file with a `GLPI_LOCAL_I18N_DIR` constant).

Reviewer: ```suggestion
Depending on your GLPI configuration, the ``_locales`` directory can be located elsewhere (if you provided a ``local_define.php`` file with a ``GLPI_LOCAL_I18N_DIR`` constant).
```

> Depending on your GLPI configuration, the ``_locales`` directory can be located elsewhere (if you provided a ``local_define.php`` file with a ``GLPI_LOCAL_I18N_DIR`` constant).
> It's possible to have multiple files, to manage several languages for example.

Reviewer: ```suggestion

.. note::
   Depending on your GLPI configuration, the ``_locales`` directory can be located elsewhere (if you provided a ``local_define.php`` file with a ``GLPI_LOCAL_I18N_DIR`` constant).
   It's possible to have multiple files, to manage several languages for example.
```

Reviewer: Either this, or split in two distinct paragraphs.

Author: ok for me

The file must be a valid `gettext` file. You can use `Poedit <https://poedit.net/>`_ to create it.

.. code-block:: po

    # Override PO file
    #
    msgid ""
    msgstr ""
    "Project-Id-Version: PACKAGE VERSION\n"
    "Report-Msgid-Bugs-To: \n"
    "POT-Creation-Date: 2019-04-13 10:43+0200\n"
    "PO-Revision-Date: 2019-04-13 10:43+0200\n"
    "Last-Translator: Automatically generated\n"
    "Language-Team: none\n"
    "Language: en_GB\n"
    "MIME-Version: 1.0\n"
    "Content-Type: text/plain; charset=UTF-8\n"
    "Content-Transfer-Encoding: 8bit\n"
    "Plural-Forms: nplurals=2; plural=(n != 1);\n"

    msgid "Login"
    msgstr "Login from local gettext"

`msgid` is the source string, `msgstr` is the translation.

> You can find the correct source sentences to use in the `locales/glpi.pot` file.

Reviewer: ```suggestion
You can find the correct source sentences to use in the ``locales/glpi.pot`` file.
```

Next you need to compile your .po file into a .mo file readable by GLPI. To do so, you can use the following command:

.. code-block:: bash

    $ cd files/_locales/core/
    $ msgfmt -o filename.mo filename.po

Finally, GLPI uses a cache system to avoid loading the .mo files at each request. You need to clear the cache to see your changes.

> To do so, go to "Setup > General", "Performance" tab and click on "Reset" button on the "Translation cache" section.

Reviewer 2: You need to be in debug mode in recent GLPI versions to see the options to reset the cache(s) here.

    msgid "Login"
    msgstr "Login from local gettext"

> `msgid` is the original string, `msgstr` is the translation.
> You can find the correct sentences to use in the `locales/glpi.pot` file.

Reviewer: ```suggestion
`msgid` is the source string, `msgstr` is the translation.
You can find the correct source sentences to use in the `locales/glpi.pot` file.
```
